engine/apps/mobile_app/serializers.py

```diff
@@ -43,3 +45,40 @@ def validate_going_oncall_notification_timing(
                 if option not in notification_timing_options:
                     raise serializers.ValidationError(detail="invalid timing options")
         return going_oncall_notification_timing
+
+
+class FCMDeviceSerializer(BaseFCMDeviceSerializer):
+    def validate(self, attrs):
+        """
+        Overrides `validate` method from BaseFCMDeviceSerializer to allow different users have same device
+        `registration_id` (multi-stack support).
+        Removed deactivating devices with the same `registration_id` during validation.
+        """
+        devices = None
+        request_method = None
+
+        if self.initial_data.get("registration_id", None):
+            if self.instance:
+                request_method = "update"
+            else:
+                request_method = "create"
+        else:
+            if self.context["request"].method in ["PUT", "PATCH"]:
```

Comment:
_nit_ consider extracting `self.context["request"]` out into a variable towards the top of this function (as it's accessed 3 times)